solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -214,16 +214,15 @@ public ModelWeight(IndexSearcher searcher, FeatureWeight[] modelFeatures,
      * Goes through all the stored feature values, and calculates the normalized
      * values for all the features that will be used for scoring.
      */
-    public void normalize() {
+    public void makeRawFeatures() {
```

Comment:
Is it possible to move this into the scoring algorithm?  The only thing it needs is allFeatureValues and allFeatureUsed.  I think the scoring algo's model features have the feature id to index into allFeatureValues.